app/models/provider/synth.rb

```diff
@@ -57,12 +57,40 @@ def fetch_exchange_rate(from:, to:, date:)
     end
   end
 
+  def fetch_exchange_rate_for_date_range(from:, to:, date_start:, date_end:)
```

Comment:
Mostly style here, but could we keep this consistent with the `fetch_security_prices` method stub?
```suggestion
  def fetch_exchange_rates(from:, to:, start_date:, end_date:)
```

Reply:
Done.